### DISCUSSION TITLE
How to create an anvil GUI

### DISCUSSION THREAD

**Discussion by bakane6030** [Minestom/Minestom#769] [Help] [answered]
Hi. I want to open an anvil GUI and extend it if I right click on an anvil. How can I do that?

**Comment by ZakShearman** [comment, marked as answer]
You can use PlayerBlockInteractEvent to detect when a block is right clicked.

You can make an anvil inventory using `Inventory inventory = new Inventory(InventoryType.ANVIL, title)`

To detect actions within the anvil you can use `InventoryPreClickEvent` and the getSlot method to identify what is clicked.

**Reply by bakane6030** [reply]
Thanks!

**Reply by bakane6030** [reply]
Although I'm not really sure how to get the items from the two anvil slots. I need to get the item in the first and second slot once they are both there. How can I do that? @ZakShearman

**Reply by emortaldev** [reply]
inventory.getItemStack()?

**Reply by MrlnHi** [reply]
Use a combination of InventoryClickEvent and InventoryPreClickEvent. At least that's what I did when creating an anvil gui and it's working pretty well
